File: JSPM.Web/Modules/Organisation/Branches/BranchesForm.cs
using Serenity.ComponentModel;

namespace JSPM.Organisation.Forms;

[FormScript("Organisation.Branches")]
[BasedOnRow(typeof(BranchesRow), CheckNames = true)]
public class BranchesForm
{
    public string BranchName { get; set; }
    public string BranchCode { get; set; }
    public int InstituteId { get; set; }
    public int DepartmentId { get; set; }
}

File: JSPM.Web/Modules/ServerTypes/Organisation/BranchesForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface BranchesForm {
    BranchName: StringEditor;
    BranchCode: StringEditor;
    InstituteId: IntegerEditor;
    DepartmentId: IntegerEditor;
}

export class BranchesForm extends PrefixedContext {
    static formKey = 'Organisation.Branches';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BranchesForm.init)  {
            BranchesForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(BranchesForm, [
                'BranchName', w0,
                'BranchCode', w0,
                'InstituteId', w1,
                'DepartmentId', w1
            ]);
        }
    }
}